pkg/cloudprovider/aws/fleet/packing/instancetype.go

```diff
@@ -0,0 +1,131 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package packing
+
+import (
+	"errors"
+	"sort"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// TODO get this information from node-selector
+var (
+	instanceTypesAvailable = []*instanceType{
+		{
+			name: "m5.8xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("32000m"),
+				v1.ResourceMemory: resource.MustParse("128Gi"),
+			},
+			utilizedCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("0"),
+				v1.ResourceMemory: resource.MustParse("0"),
+			},
+		},
+		{
+			name: "m5.2xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("8000m"),
+				v1.ResourceMemory: resource.MustParse("32Gi"),
+			},
+			utilizedCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("0"),
+				v1.ResourceMemory: resource.MustParse("0"),
+			},
+		},
+		{
+			name: "m5.xlarge",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("4000m"),
+				v1.ResourceMemory: resource.MustParse("16Gi"),
+			},
+			utilizedCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("0"),
+				v1.ResourceMemory: resource.MustParse("0"),
+			},
+		},
+		{
+			name: "m5.large",
+			totalCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("2"),
+				v1.ResourceMemory: resource.MustParse("8Gi"),
+			},
+			utilizedCapacity: v1.ResourceList{
+				v1.ResourceCPU:    resource.MustParse("0"),
+				v1.ResourceMemory: resource.MustParse("0"),
+			},
+		},
+	}
+)
+
+var InsufficentCapacityErr = errors.New("insufficient capacity")
+
+type instanceType struct {
+	name             string
+	utilizedCapacity v1.ResourceList
+	totalCapacity    v1.ResourceList
```

Comment:
I'm not sure that utilizedCapacity and totalCapacity should be the same types? Seems like utilization should be a percentage (unless we need to model that somehow utilization can go over 100%?)

Reply:
We need to calculate 3 utilization metrics, CPU, memory and pods count. Percentage does makes sense for CPU and memory, but not sure if it works for pods count? 
Also, having it as a ResourceList makes it possible for adding more utilization metrics in the future.

Reply:
Sorry I was probably unclear, I didn't mean the idea of percentages was necessarily bad, I meant that using the same percentage value for all resources might not be flexible enough? Like a customer might want to scale based on CPU utilization being 80% or not, and memory utilization being 40% or not? If I understand the code correctly, they'd be unable to set different percentages for different types of resources.

Reply:
I might've misunderstood what ResourceLists are ... resolving for now.